File: service/app/src/test/java/io/miragon/blueprint/application/service/IssueInsurancePolicyServiceTest.java
package io.miragon.blueprint.application.service;

import static io.miragon.blueprint.domain.leasing.TestObjectBuilder.testLeasingApplication;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.verifyNoMoreInteractions;

import io.miragon.blueprint.application.port.outbound.InsurancePort;
import io.miragon.blueprint.domain.leasing.LeasingApplication;
import org.junit.jupiter.api.Test;

class IssueInsurancePolicyServiceTest {

    private final InsurancePort insurance = mock(InsurancePort.class);
    private final IssueInsurancePolicyService underTest = new IssueInsurancePolicyService(insurance);

    @Test
    void issuePolicyDelegatesToTheInsurer() {
        // given: an eligible application
        LeasingApplication application = testLeasingApplication().build();
        // when: the policy is issued
        underTest.issuePolicy(application.id());
        // then: the insurance out-port binds the policy
        verify(insurance).issuePolicy(application.id());
        verifyNoMoreInteractions(insurance);
    }
}

File: service/app/src/test/kotlin/io/miragon/blueprint/application/service/IssueInsurancePolicyServiceTest.kt
package io.miragon.blueprint.application.service

import io.miragon.blueprint.application.port.outbound.InsurancePort
import io.miragon.blueprint.domain.leasing.testLeasingApplication
import io.mockk.Runs
import io.mockk.confirmVerified
import io.mockk.every
import io.mockk.just
import io.mockk.mockk
import io.mockk.verify
import org.junit.jupiter.api.Test

class IssueInsurancePolicyServiceTest {

    private val insurance = mockk<InsurancePort>()
    private val underTest = IssueInsurancePolicyService(insurance = insurance)

    @Test
    fun `issuePolicy delegates to the insurer`() {
        // given: an eligible application
        val application = testLeasingApplication()
        every { insurance.issuePolicy(application.id) } just Runs
        // when: the policy is issued
        underTest.issuePolicy(application.id)
        // then: the insurance out-port binds the policy
        verify { insurance.issuePolicy(application.id) }
        confirmVerified(insurance)
    }
}
